### PR TITLE
SO-4945: Improve Suggestions API

### PR DESCRIPTION
**Pull request by molnarlaura**

### CODE REVIEW

File: core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/suggest/SuggestRestService.java [hunk 1]
```diff
@@ -61,11 +62,18 @@
...
-				.setPreferredDisplay(params.getPreferredDisplay())
-				.filterByTerm(params.getTerm())
+		ConceptSuggestionRequestBuilder requestBuilder = CodeSystemRequests.prepareSuggestConcepts()
+						.setLimit(params.getLimit())
+						.setLocales(Strings.isNullOrEmpty(params.getAcceptLanguage()) ? acceptLanguage : params.getAcceptLanguage())
+						.setPreferredDisplay(params.getPreferredDisplay())
+						.setMinOccurrenceCount(params.getMinOccurrenceCount())
+						.filterByTerm(params.getTerm());
+		
+		if (params.getMinOccurrenceCount() != null) {
```
Comment (cmark) [inline core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/suggest/SuggestRestService.java]: Please set the `minOccurenceCount` only once.

File: core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/suggest/SuggestRestService.java [hunk 2]
```diff
@@ -86,11 +94,18 @@
...
-				.setPreferredDisplay(body.getPreferredDisplay())
-				.filterByTerm(body.getTerm())
+		ConceptSuggestionRequestBuilder requestBuilder = CodeSystemRequests.prepareSuggestConcepts()
+						.setLimit(body.getLimit())
+						.setLocales(Strings.isNullOrEmpty(body.getAcceptLanguage()) ? acceptLanguage : body.getAcceptLanguage())
+						.setPreferredDisplay(body.getPreferredDisplay())
+						.setMinOccurrenceCount(body.getMinOccurrenceCount())
+						.filterByTerm(body.getTerm());
+		
+		if (body.getMinOccurrenceCount() != null) {
```
Comment (cmark) [inline core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/suggest/SuggestRestService.java]: Same here.

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptSuggestionRequest.java [hunk 3]
```diff
@@ -88,7 +88,7 @@ protected Suggestions doExecute(BranchContext context) throws IOException {
 		TermFilter termFilter;
 		
 		if (containsKey(TERM)) {
-			termFilter = TermFilter.defaultTermMatch(getString(TERM)).withIgnoreStopwords();
+			termFilter = TermFilter.minTermMatch(getString(TERM), minOccurrenceCount).withIgnoreStopwords();
```
Comment (cmark) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptSuggestionRequest.java]: Convert the `minOccurenceCount` to a key and make sure it is set to the default three value when it is not set and base term is not given.

**Review by cmark — CHANGES_REQUESTED** [review]

File: core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/suggest/SuggestRestService.java (hunk 1 above)
Comment (molnarlaura) [reply core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/suggest/SuggestRestService.java]: Done

File: core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/suggest/SuggestRestService.java (hunk 2 above)
Comment (molnarlaura) [reply core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/suggest/SuggestRestService.java]: Done

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptSuggestionRequest.java (hunk 3 above)
Comment (molnarlaura) [reply core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptSuggestionRequest.java]: Done

**Review by cmark — APPROVED** [review]
LGTM!

**Pull request merged**